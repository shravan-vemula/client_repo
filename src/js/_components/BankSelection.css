body{
    background-color: #fafafb !important;
}

.bankSelection{
    width: 100%;
    height: 100%;
    background-color: #fafafb;
}

.bankSelectBar{
    background-color: white !important;
    color: black !important;
}

.logoBtracker{
    width: 8%;
}

#logo{
    cursor: pointer;
}

.BtrackerIcon{
    width:100%
}

.logoutButton{
    position: relative;
    left: 80%;
}

.MuiToolbar-root{
    height: 50px !important;
    min-height: 40px !important;
}

.bankSelectionLayout1{
    height: 500px;
    width: 60%;
    padding: 0%;
    margin-top: 5%;
    margin-left: 14%;
    border-right: 2px solid;
    border-right-color: #d4d7dc;
    display: inline-block;
    float: left;
    text-align: left;
}

.bankSelectionHeader{

    font-size: 28px ;
    font-weight: 200;

}

.bankSelectionHeader2{
    font-size: 15px;
    font-weight: 400;
}

#bank-tiles{
    margin-top: 5%;
    width: 70%;
}

.optionalSelection{
    color: #BABEC5;
    margin-top: 5%;
}

.sideText1{
    float: right;
    margin-top: 8%;
    margin-right: 6%;
    text-align: left;
}

.sideText2{
    margin-top: 1% !important;
    text-align: left;
    margin-right: 8% !important;
}
.sideText3{
    
    margin-right: 8% !important;
}

.lockIcon{
    display: inline-block;
    float: left;
}

.IconStyle{
    position: relative;
    top: 50%;
    margin-left: 30%;
    margin-top: 40%;
}

.SideTextSteps2{
    margin-top: 0px !important;
}

.sideText2 li{
    list-style-type: none;
}

.text2{
    display: inline-block;
}

.sideText1 h1{
    font-size: 15px;
    line-height: 20px;
    font-weight: 500;
}

.sideText1 li {
    font-size: 14px;
    font-weight: 200;
    color: slategray;
}

.SideTextSteps{
    padding-left: 15px !important;

}

.SideTextSteps li{
    margin: 2%;

}