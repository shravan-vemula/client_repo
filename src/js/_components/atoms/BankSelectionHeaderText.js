import React from 'react'

function BankSelectionHeaderText() {
    return (
        <>
            <div id="bank-selection-header" className="bankSelectionHeader">
                <div id="bank-selection-header-text-1"className="bankSelectionHeader1" >
                    <p>See all your money in one place</p>
                    </div>
                <div id="bank-selection-header-text-1"className="bankSelectionHeader2" >
                    <p>Connect your <b>banks, bills</b> and <b>credit cards</b> to get insights into your spending. We even allow <br /> 
                        accounts like Paytm and PhonePe.</p>
                </div>    
            </div>
        </>
    )
}

export default BankSelectionHeaderText
