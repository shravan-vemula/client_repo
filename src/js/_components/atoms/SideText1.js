import React from 'react'

export default function SideText1() {
    return (
        <>
        <div id="side-text-1" className="sideText1">
           <h1> Three easy steps to get <br /> started with Mint </h1>
            <ul id="steps" className="SideTextSteps">
            <li>
            Find your bank<br />
            or credit card
            </li>
            <li>
            Connect it
            securely to Mint
            </li>
            <li>
            Let Mint do
            the rest
            </li> 

            </ul>
        </div>
        </>
    )
}
