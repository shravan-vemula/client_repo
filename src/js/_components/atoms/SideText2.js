import React from 'react'
import EnhancedEncryptionIcon from '@material-ui/icons/EnhancedEncryption';

export default function SideText2() {
    return (
        <>
        <div id="side-text-1" className="sideText1 sideText2">
           <h1> Serious about security</h1>
           <div id="lock-icon" className="lockIcon">
            <EnhancedEncryptionIcon className="IconStyle"/>
            </div>
            <div id="text-2" className="text2">
            <ul id="steps" className="SideTextSteps SideTextSteps2">
                <li>
                We are Norton <br/> certified to help <br/> ensure security for <br />sensitive data
                </li>
            </ul>
            </div>
        </div>
        </>
    )
}
