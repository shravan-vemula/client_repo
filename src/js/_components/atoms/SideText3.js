import React from 'react'
import SecurityIcon from '@material-ui/icons/Security';

export default function SideText3() {
    return (
        <>
        <div id="side-text-1" className="sideText1 sideText2 sideText3">
           <h1>Trusted by millions</h1>
           <div id="lock-icon" className="lockIcon">
            <SecurityIcon className="IconStyle"/>
            </div>
            <div id="text-2" className="text2">
            <ul id="steps" className="SideTextSteps SideTextSteps2">
                <li>
                Over 300 million <br/> accounts added
                </li>
            </ul>
            </div>
        </div>
        </>
    )
}
