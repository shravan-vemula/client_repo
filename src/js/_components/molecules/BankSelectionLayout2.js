import React from 'react'
import SideText from '../organisms/SideText'

function BankSelectionLayout2() {
    return (
        <>
        <div id="bank-selection-layout-2" className="BankSelectionLayout2">
        <SideText />

        </div>
        </>
    )
}

export default BankSelectionLayout2
