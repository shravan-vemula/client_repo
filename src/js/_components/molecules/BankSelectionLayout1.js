import React from 'react'
import BankSelectionHeaderText from '../atoms/BankSelectionHeaderText';
import SearchBar from '../atoms/SearchBar';
import BankTiles from '../organisms/BankTiles'


function BankSelectionLayout1() {
    return (
        <>

            <div id="bank-selection-layout-1" className="bankSelectionLayout1">
                <BankSelectionHeaderText />
                <SearchBar />
            <div id="optional-selection" className="optionalSelection">
            Or select from popular ones    
            </div>
                <BankTiles />
            </div>
        </>
    )
}

export default BankSelectionLayout1
