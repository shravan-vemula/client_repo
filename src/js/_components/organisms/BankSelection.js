import React, { Component } from 'react'
import BankSelectionLayout1 from '../molecules/BankSelectionLayout1';
import BankSelectionLayout2 from '../molecules/BankSelectionLayout2';
import NavBar from './Navbar';
import 'js/_components/BankSelection.css';
import { Switch,Route} from 'react-router-dom';
import LoginPage from 'js/_components/organisms/LoginPage';
import BankPageRedirect from '../molecules/BankPageRedirect';

export class BankSelection extends Component {
    render() {
        return (
        <>
            <div id="nav-bar" className="navbar"  style={{fontFamily:"Inter,sans-serif"}}>
            <NavBar />
            </div>
            <div id="bank-selection" className="bankSelection"  style={{fontFamily:"Inter,sans-serif"}}>
                
                    <Switch>

                    <Route path="/SelectBank/BankTiles" exact component={LoginPage} > 
                        <BankSelectionLayout1 />
                    </Route>
                        <Route path="/SelectBank/LoginPage" exact component={LoginPage} >
                            <LoginPage />    
                        </Route>
                        <Route path="/SelectBank/RedirectLogin" exact component={BankPageRedirect} >
                            <BankPageRedirect />    
                        </Route>

                    </Switch>
                
                <BankSelectionLayout2 />
            </div>
        </>
        )
    }
}

export default BankSelection
