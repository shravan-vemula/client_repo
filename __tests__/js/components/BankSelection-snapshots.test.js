import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import {BrowserRouter as Router} from 'react-router-dom';
import SideText1 from 'js/_components/atoms/SideText1';
import SideText from 'js/_components/organisms/SideText';
import BankSelectionLayout1 from 'js/_components/molecules/BankSelectionLayout1';
import BankSelection from 'js/_components/organisms/BankSelection';
import BankSelectionLayout2 from 'js/_components/molecules/BankSelectionLayout2';
import SideText2 from 'js/_components/atoms/SideText2';
import SideText3 from 'js/_components/atoms/SideText3';
import BankSelectionHeaderText from 'js/_components/atoms/BankSelectionHeaderText';
import BankTiles from 'js/_components/organisms/BankTiles';

describe('FormLayout', () =>{
    test('renders form layout', () =>{
        render(<Router><BankSelectionLayout1 /></Router>);
        expect(screen.getByText('See all your money in one place')).toBe;
        render(<Router><SideText1 /></Router>);
        expect(screen.getByText('Three easy steps to get started with Mint')).toBe;
        render(<Router><SideText /></Router>);
        expect(screen.getByText('Serious about security')).toBe;
        expect(screen.getByText('Over 300 million accounts added')).toBe;

    })


    it("renders BankSelection correctly", () => {
        const wrapper = shallow(
            <BankSelection /> 
        );
        expect(wrapper).toMatchSnapshot();
      });


    it("renders BankSelectionLayout1 correctly", () => {
        const wrapper = shallow(
            <BankSelectionLayout1 /> 
        );
        expect(wrapper).toMatchSnapshot();
      });
      
      it("renders BankSelectionLayout2 correctly", () => {
        const wrapper = shallow(
            <BankSelectionLayout2 /> 
        );
        expect(wrapper).toMatchSnapshot();
      });

      it("renders SideText correctly", () => {
        const wrapper = shallow(
            <SideText /> 
        );
        expect(wrapper).toMatchSnapshot();
      });
      it("renders SideText1 correctly", () => {
        const wrapper = shallow(
            <SideText1 /> 
        );
        expect(wrapper).toMatchSnapshot();
      });

      it("renders SideText2 correctly", () => {
        const wrapper = shallow(
            <SideText2 /> 
        );
        expect(wrapper).toMatchSnapshot();
      });

      it("renders SideText3 correctly", () => {
        const wrapper = shallow(
            <SideText3 /> 
        );
        expect(wrapper).toMatchSnapshot();
      });

      it("renders BankSelectionHeaderText correctly", () => {
        const wrapper = shallow(
            <BankSelectionHeaderText /> 
        );
        expect(wrapper).toMatchSnapshot();
      });

      it('Bank tiles test',() =>{
        const {getByTestId} = render(<Router><BankTiles /> </Router>)
        expect(screen.getByTestId('tileCLick')).toBe
        fireEvent.click(getByTestId('tileCLick'))
        
      })

}); 