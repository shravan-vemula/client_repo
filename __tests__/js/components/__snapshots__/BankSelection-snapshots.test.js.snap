// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLayout renders BankSelection correctly 1`] = `ShallowWrapper {}`;

exports[`FormLayout renders BankSelectionHeaderText correctly 1`] = `ShallowWrapper {}`;

exports[`FormLayout renders BankSelectionLayout1 correctly 1`] = `ShallowWrapper {}`;

exports[`FormLayout renders BankSelectionLayout2 correctly 1`] = `ShallowWrapper {}`;

exports[`FormLayout renders SideText correctly 1`] = `ShallowWrapper {}`;

exports[`FormLayout renders SideText1 correctly 1`] = `ShallowWrapper {}`;

exports[`FormLayout renders SideText2 correctly 1`] = `ShallowWrapper {}`;

exports[`FormLayout renders SideText3 correctly 1`] = `ShallowWrapper {}`;
